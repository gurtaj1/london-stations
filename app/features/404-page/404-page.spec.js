import React from 'react';
import { shallow } from 'enzyme';
import Page404 from 'features/404-page/404-page';

describe('<Page404 />', () => {
  it('should export a class', () => {
    expect(Page404).toBeDefined();
  });

  it('should render as expected, without crashing', () => {
    const wrapper = shallow(<Page404 />);

    expect(wrapper).toMatchSnapshot();
  });
});
