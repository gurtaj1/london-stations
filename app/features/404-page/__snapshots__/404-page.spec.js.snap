// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page404 /> should render as expected, without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Page404 />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="error-page__404"
        >
          404
        </div>,
        <div
          className="error-page__main-msg"
        >
          This page does not exist.
        </div>,
        <div
          className="error-page__description"
        >
          The page you're looking for has either been deleted or never existed.
          <br />
          We apologise for any inconvenience.
        </div>,
        <div
          className="error-page__button"
        >
          <Link
            className="error-page__home-button"
            replace={false}
            to="/"
          >
            Go back to home
          </Link>
        </div>,
      ],
      "className": "error-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404",
          "className": "error-page__404",
        },
        "ref": null,
        "rendered": "404",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This page does not exist.",
          "className": "error-page__main-msg",
        },
        "ref": null,
        "rendered": "This page does not exist.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "The page you're looking for has either been deleted or never existed.",
            <br />,
            "We apologise for any inconvenience.",
          ],
          "className": "error-page__description",
        },
        "ref": null,
        "rendered": Array [
          "The page you're looking for has either been deleted or never existed.",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "We apologise for any inconvenience.",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="error-page__home-button"
            replace={false}
            to="/"
          >
            Go back to home
          </Link>,
          "className": "error-page__button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Go back to home",
            "className": "error-page__home-button",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Go back to home",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="error-page__404"
          >
            404
          </div>,
          <div
            className="error-page__main-msg"
          >
            This page does not exist.
          </div>,
          <div
            className="error-page__description"
          >
            The page you're looking for has either been deleted or never existed.
            <br />
            We apologise for any inconvenience.
          </div>,
          <div
            className="error-page__button"
          >
            <Link
              className="error-page__home-button"
              replace={false}
              to="/"
            >
              Go back to home
            </Link>
          </div>,
        ],
        "className": "error-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404",
            "className": "error-page__404",
          },
          "ref": null,
          "rendered": "404",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This page does not exist.",
            "className": "error-page__main-msg",
          },
          "ref": null,
          "rendered": "This page does not exist.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "The page you're looking for has either been deleted or never existed.",
              <br />,
              "We apologise for any inconvenience.",
            ],
            "className": "error-page__description",
          },
          "ref": null,
          "rendered": Array [
            "The page you're looking for has either been deleted or never existed.",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "We apologise for any inconvenience.",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="error-page__home-button"
              replace={false}
              to="/"
            >
              Go back to home
            </Link>,
            "className": "error-page__button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Go back to home",
              "className": "error-page__home-button",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Go back to home",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
